### PR TITLE
[Rust] Add Axum

### PR DESCRIPTION
**Pull request by SabrinaJewson**
> `axum` is a web application framework that focuses on ergonomics and modularity.

cc @davidpdrsn @jplatte

### CODE REVIEW

File: rust/axum/Cargo.toml
Comment (waghanza) [inline rust/axum/Cargo.toml:8]: seems too strict for me, I think 
```suggestion
axum = "~0.4"
```

should be better, to make sure we are up-to-date but using 0.4.x (their is a bot taking care of updates here)

**Review by waghanza — CHANGES_REQUESTED** [review]

**Comment by waghanza** [comment]
Hi @SabrinaJewson,

Thanks for your contribution. 

I'm waiting for the explicit consent of any codeowner (of `axum`) before merging it.

I'll be glad if I can ping someone on each minor update

File: rust/axum/Cargo.toml
Comment (jplatte) [reply rust/axum/Cargo.toml:8]: Cargo treats `0.4.3` as `^0.4.3`, not `=0.4.3`. `~0.4` would additionally allow *older* versions than 0.4.3, newer ones would be chosen with either version bound (when there is no lockfile or when `cargo update` is run).

**Comment by davidpdrsn** [comment]
I'm the creator of axum 😊 This looks good to me!

File: rust/axum/Cargo.toml
Comment (waghanza) [reply rust/axum/Cargo.toml:8]: I assume then this will keep into 0.4.x, that the he only thing to notice here.

I think better to stick to `rust` best practices, I let then you decide guys (I'm more a `ruby` dev and we do not have `^` constraints)

Comment (jplatte) [reply rust/axum/Cargo.toml:8]: Yeah, `^0.x.y` for some x > 0 means `>=0.x.y && <0.(x+1).0` in Rust & NodeJS land.

**Review by waghanza — APPROVED** [review]

**Comment by waghanza** [comment]
@SabrinaJewson `axum` is in https://web-frameworks-benchmark.netlify.app/result

@davidpdrsn @jplatte May I notify one of you (or both) or minor updates ? It could help keeping a quality project

**Comment by davidpdrsn** [comment]
> May I notify one of you (or both) or minor updates ? It could help keeping a quality project

Yep just ping me!

**Pull request merged**
